packages/node/src/dev-server.ts

```diff
@@ -136,77 +147,150 @@ async function serializeRequest(message: IncomingMessage) {
   });
 }
 
-async function createEdgeEventHandler(
-  entrypoint: string
-): Promise<(request: IncomingMessage) => Promise<VercelProxyResponse>> {
-  const buildResult = await ncc(entrypoint, { target: 'es2022' });
-  const userCode = buildResult.code;
-
-  const initialCode = `
-    ${userCode};
-
-    addEventListener('fetch', async (event) => {
-      let serializedRequest = await event.request.text();
-      let requestDetails = JSON.parse(serializedRequest);
+async function compileUserCode(entrypoint: string) {
+  try {
+    const buildResult = await ncc(entrypoint, {
+      target: 'es2022',
+    });
+    const userCode = buildResult.code;
+
+    return `
+      ${userCode};
+
+      addEventListener('fetch', async (event) => {
+        try {
+          let serializedRequest = await event.request.text();
+          let requestDetails = JSON.parse(serializedRequest);
+
+          let body;
+
+          if (requestDetails.method !== 'GET' && requestDetails.method !== 'HEAD') {
+            body = Uint8Array.from(atob(requestDetails.body), c => c.charCodeAt(0));
+          }
+
+          let requestUrl = requestDetails.headers['x-forwarded-proto'] + '://' + requestDetails.headers['x-forwarded-host'] + requestDetails.url;
+
+          let request = new Request(requestUrl, {
+            headers: requestDetails.headers,
+            method: requestDetails.method,
+            body: body
+          });
+
+          event.request = request;
+
+          let edgeHandler = module.exports.default;
+          if (!edgeHandler) {
+            throw new Error('No default export was found. Add a default export to handle requests.');
```

Comment:
@leerob is there a good link we can use here to point users to a place where they can read up on general edge functions? I know things are shifting around.

Reply:
Added in: https://github.com/vercel/vercel/pull/8048